src/dashboard/tests/test_helpers.py

```diff
@@ -0,0 +1,159 @@
+        transfer_name="mets_transfer", sip_uuid=sip_uuid
+    )
+    mock.assert_called_once_with(mets_file_path, "r")
+    assert response.get("Content-Type") == "application/xml"
+    assert response.get("Content-Disposition") == "attachment; filename={};".format(
+        mets_file
+    )
+    response_text = ""
+    for response_char in response.streaming_content:
+        response_text = "{}{}".format(response_text, response_char)
+    assert response_text == METS
+
+
+def test_stream_pointer_from_storage_unsuccessful(mocker):
+    pointer_url = "http://archivematica-storage-service:8000/api/v2/file/44444444-4444-4444-4444-444444444444/pointer_file"
+    custom_error_message = "Unable to retrieve AIP pointer file from Storage Service"
+    mock_response = requests.Response()
+    mock_response.status_code = 503
+    mocker.patch.object(requests, "get", return_value=mock_response)
+    response = helpers.stream_file_from_storage_service(
+        pointer_url, error_message=custom_error_message
+    )
+    err = json.loads(response.content)
+    assert err.get("success") is False
+    assert custom_error_message in err.get("message")
+    assert response.status_code == 400
+
+
+def test_stream_pointer_from_storage_successful(mocker):
+    METS_PTR = """<?xml version='1.0' encoding='UTF-8'?>
+    <mets:mets xmlns:mets="http://www.loc.gov/METS/" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.loc.gov/METS/ http://www.loc.gov/standards/mets/version1121/mets.xsd">
+        <mets:metsHdr CREATEDATE="2020-01-21T15:43:07"/>
+    </mets:mets>
+    """
+    sip_uuid = "55555555-5555-5555-5555-555555555555"
+    pointer_url = "http://archivematica-storage-service:8000/api/v2/file/{}/pointer_file".format(
+        sip_uuid
+    )
+    pointer_file = "pointer.{}.xml".format(sip_uuid)
+    content_disposition = 'attachment; filename="{}"'.format(pointer_file)
+    mock_response = requests.Response()
+    mock_response.status_code = 200
+    mock_response.headers = {
+        "content-type": "application/xml",
+        "content-disposition": content_disposition,
+    }
+    mock_response._content = METS_PTR
+    mocker.patch.object(requests, "get", return_value=mock_response)
+    response = helpers.stream_file_from_storage_service(pointer_url)
+    assert response.status_code == 200
+    assert response.get("Content-Type") == "application/xml"
+    assert response.get("Content-Disposition") == content_disposition
+    response_text = ""
+    for response_char in response.streaming_content:
+        response_text = "{}{}".format(response_text, response_char)
+    assert response_text == METS_PTR
+    # Make the request again, but ask to view in the browser, i.e. inline.
+    response = helpers.stream_file_from_storage_service(pointer_url, preview_file=True)
+    assert response.get("Content-Disposition") == "inline"
+    assert response_text == METS_PTR
```

Comment:
Does this assertion have any additional value than the one a few lines above?

Reply:
It's simply because I make a request for a second time that I want the tests to guarantee that the response is identical a second time as well. It might change for example if someone had written `preview_file=...` to take the code through an entirely different pathway that affected the way the response was returned.

Reply:
In that case shouldn't `response_text` be updated as well after the second request is made?

Reply:
I see the problem, yes, you are right, I need to generate a new `response_text` here from the streaming content, which makes perfect sense! Good catch!